src/send/saga.ts

```diff
@@ -151,6 +151,7 @@ export function* buildAndSendPayment(
     feeInfo
   )
 
+  // TODO: We need to add the transaction hash in order to be able to watch if the transaction fails.
```

Comment:
how should this TODO be addressed? i think it's not possible to get the tx hash ahead of time sometimes, which is why i think we should remove any pending tx's that don't have tx hashes at the start of new app sessions

Reply:
This is a good question. I'm not sure how, but as far as I understand the transaction hash can be obtained before sending the transaction to the blockchain.
I think we should do the effort / investigation to understand if this is feasible in all cases, and if so, we should force the pending transactions to have a transaction hash.

Reply:
it's my understanding that for the swap flow using viem, we have the tx hash before sending the tx because we prepare the transactions upfront. but for the contract kit flows and the viem send flow, we do not. i'm not sure that it's possible to get this with contractkit upfront (and tbh not sure that it is worth the effort since we want to move away from it - so it seems like timing wise, it's not going to be worth investigating this topic for this task?)